language/move-prover/tests/sources/serialize_model.move

```diff
@@ -0,0 +1,84 @@
+// dep: tests/sources/stdlib/modules/lcs.move
+
+module TestLCS {
+
+    use 0x0::LCS;
+
+    // *** Not sure how to have a generic type var for whole module?
+
+    // serialize tests
+
+    fun lcs_test1<Thing>(v1: &Thing, v2: &Thing): (vector<u8>, vector<u8>)
+    {
+        let s1 = LCS::to_bytes(v1);
+        let s2 = LCS::to_bytes(v2);
+        (s1, s2)
+    }
+    spec fun lcs_test1 {
+        aborts_if false;
+        ensures result_1 == result_2 ==> v1 == v2;
+        ensures v1 == v2 ==> result_1 == result_2;
+        // it knows result is vector<u8>
+        ensures len(result_1) > 0 ==> result_1[0] <= max_u8();
+    }
+
+    // FIXME: Commented out because prelude "ensures" about lengths are wrong.
```

Comment:
Is there an issue or a description somewhere what is wrong?

Reply:
I will probably file one, but it wasn't high priority.  This is related to my answer to. your question about the length of bool.  I thought we knew the lengths of U8, U64, U128, but we don't really know the types.  They're all Integer, and $IsValidU8 etc. just check the bounds on the integer, which may not be the same as the length of the data encoding.  We need a little more bookkeeping to keep track of the length in bytes of the actual data.  I left this in so I wouldn't have to regenerate it if and when we fix that.

Reply:
Ping on this comment. I just don't understand what is "wrong" here and why you left a "FIXME".